api/base/utils.py

```diff
@@ -22,3 +27,31 @@ def get_object_or_404(model_cls, query_or_pk):
         return model_cls.find_one(query)
     except NoResultsFound:
         raise NotFound
+
+
+def waterbutler_url_for(request_type, provider, path, node_id, token, obj_args=None, **query):
```

Comment:
Isn't that what you asked for in https://github.com/CenterForOpenScience/osf.io/pull/2631/files#r29700067 ? Or are you asking for something slightly different.